src/sql/workbench/contrib/editData/browser/editDataEditor.ts

```diff
@@ -290,12 +292,28 @@ export class EditDataEditor extends BaseEditor {
 			this._sash = this._register(new HorizontalFlexibleSash(parentElement, this._minEditorSize));
 			this._setSashDimension();
 
-			this._register(this._sash.onPositionChange(position => this._doLayout()));
+			this._register(this._sash.onPositionChange(position => this._positionCheck(position)));
 		}
 
 		this._sash.show();
 	}
 
+	/**
+	 * Checks if the current position of the sash doesn't cover the first row of the table.
+	 * If large enough, do layout. Else if too small, hide result editor to prevent crashing.
```

Comment:
What happens is that, if the first row of the grid is overlapped in any way, it will crash unexpectedly (with the sash dragged over it), so it will have to be hidden to prevent it from crashing.